src/modules/tools/jogger/Jogger.h

```diff
@@ -0,0 +1,59 @@
+
+/*
+this file is part of smoothie (http://smoothieware.org/). the motion control part is heavily based on grbl (https://github.com/simen/grbl).
+smoothie is free software: you can redistribute it and/or modify it under the terms of the gnu general public license as published by the free software foundation, either version 3 of the license, or (at your option) any later version.
+smoothie is distributed in the hope that it will be useful, but without any warranty; without even the implied warranty of merchantability or fitness for a particular purpose. see the gnu general public license for more details.
+you should have received a copy of the gnu general public license along with smoothie. if not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef JOGGER_H
+#define JOGGER_H
+
+#define NUM_JOG_AXES 3
+
+#include "Module.h"
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+class Jogger : public Module {
+public:
+    Jogger();
+
+    void on_module_loaded();
+    void on_config_reload(void* argument);
+    void on_gcode_received(void* argument);
+    uint32_t update_tick(uint32_t dummy);
+    void on_main_loop(void* argument);
+
+private:
+    float get_speed(float pos); //function to return a speed given a unitless joystick reading
+
+    float max_speed = 10.0f; //maximum allowed speed (defaults to slow 10 mm/min)
+    float dead_zone = 0.002f; //distance from 0 in which small joystick movement is ignored
+    float nonlinearity = 1.0f; //nonlinearity parameter for scaling of joystick values
+
+    float position[NUM_JOG_AXES] = {}; //keeps track of the joystick positions for each axis
```

Comment:
I do not believe this will initialize the array to 0. there is a special GCC syntax for that or do it in the constructor.